daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SValue.scala

```diff
@@ -167,6 +167,7 @@ object SValue {
 
     def apply(xs: Iterator[(SValue, SValue)]): SGenMap = {
       type O[_] = TreeMap[SValue, SValue]
+      xs.foreach { case (k, _) => comparable(k) }
```

Comment:
Maybe we can avoid iterating the collection twice by side-effectful mapping with `breakOut`?

Reply:
In the interests of forward-compatibility:

```suggestion
      xs.map { case p @ (k, _) => comparable(k); p }.to[O]
```